FormalConjectures/Other/ReedOmegaDeltaChi.lean

```diff
@@ -0,0 +1,71 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Reed's omega, delta, and chi conjecture
+
+*Reference:* (http://www.openproblemgarden.org/op/reeds_omega_delta_and_chi_conjecture)
+
+*Reference* (https://mathoverflow.net/questions/37923)
+-/
+
+namespace ReedOmegaDeltaChi
+
+open Classical
+
+noncomputable
+def maxDegree {V : Type} (G : SimpleGraph V) : ℕ∞ :=
+  if h : ∀ v, Finite (G.neighborSet v) then
+    have : ∀ v, Fintype (G.neighborSet v) :=
+      fun v => @Fintype.ofFinite _ (h v)
+    ⨆ v, G.degree v
+  else
+    ⊤
```

Comment:
Is this givin the junk value of "0" for a graph that has Finite degree for each node, but infinitely many nodes, and nodes of arbitrarily high degree? Should it be \top in that case as well. 
Perhaps more something like
```
noncomputable
def maxDegree {V : Type} (G : SimpleGraph V) : ℕ∞ :=
  ⨆ v, if h : (G.neighborSet v).Finite then
    have : Fintype (G.neighborSet v) := by exact h.fintype
    (G.degree v : ℕ∞)
  else
    ⊤
```
and then make sure `maxCliqueSize` is not depending on that junk value?

Reply:
How would I actually check if this gives `0`? I had the case of unbounded degree in mind when writing this, and I thought that the supremum would be \top in this case, since the return type is N_\infty. Maybe it would be good to include some kind of unit test here. Like define a graph with finite but unbounded degrees and prove that the `maxDegree` is `\top`.